src/components/CriteriaBuilder/CriteriaBuilder.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { FC, memo, useCallback, useEffect, useState, ChangeEvent } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPlus, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { IconButton, TextField } from '@material-ui/core';
+
+import DropDown from 'components/elements/DropDown';
+import { useBuildCriteriaContext } from 'components/BuildCriteriaProvider';
+import useStyles from './styles';
+
+const CriteriaBuilder: FC = () => {
+  const styles = useStyles();
+  const { setBuildCriteriaCql } = useBuildCriteriaContext();
+  const [selectedElement, setSelectedElement] = useState<string>('');
+  const [selectedDemoElement, setSelectedDemoElement] = useState<string>('');
+  const [gender, setGender] = useState<string>('');
+  const [minimumAge, setMinimumAge] = useState<number>(0);
+  const [maximumAge, setMaximumAge] = useState<number>(0);
+  const elementOptions = [{ value: 'Demographics', label: 'Demographics' }];
+  const demoElementOptions = [
+    { value: 'Age Range', label: 'Age Range' },
+    { value: 'Gender', label: 'Gender' }
+  ];
+  const genderOptions = [
+    {
+      label: 'Male',
+      value: 'Male'
+    },
+    {
+      label: 'Female',
+      value: 'Female'
+    },
+    {
+      label: 'Other',
+      value: 'Other'
+    },
+    {
+      label: 'Unknown',
+      value: 'Unknown'
+    }
+  ];
+
+  const onElementSelected = useCallback((event: ChangeEvent<{ value: string }>): void => {
+    setSelectedElement(event?.target.value || '');
+  }, []);
+
+  const onDemoElementSelected = useCallback((event: ChangeEvent<{ value: string }>): void => {
+    setSelectedDemoElement(event?.target.value || '');
+  }, []);
+
+  const onGenderSelected = useCallback((event: ChangeEvent<{ value: string }>): void => {
+    setGender(event?.target.value || '');
+  }, []);
+
+  const onMinimumAgeChange = useCallback((event: ChangeEvent<{ value: string }>): void => {
+    setMinimumAge(parseInt(event?.target.value) || 0);
+  }, []);
+
+  const onMaximumAgeChange = useCallback((event: ChangeEvent<{ value: string }>): void => {
+    setMaximumAge(parseInt(event?.target.value) || 0);
+  }, []);
+
+  const resetElements = useCallback(() => {
+    setSelectedElement('');
+    setSelectedDemoElement('');
+    setGender('');
+    setMinimumAge(0);
+    setMaximumAge(0);
+  }, []);
+
+  const genderString = `The patient's gender is ${gender}`;
+  const ageRangeString = `The patient's age is between ${minimumAge} years and ${maximumAge} years`;
+
+  useEffect(() => {
+    const cql = `AgeInYears() >= ${minimumAge} and AgeInYears() < ${maximumAge}`;
+    if (minimumAge >= 0 && maximumAge > 0) {
```

Comment:
Do we need to ensure that minimumAge <= maximumAge as well?

Reply:
addressed